network/test/testUtil.go

```diff
@@ -262,11 +265,14 @@ func GenerateIDsAndMiddlewares(t *testing.T,
 }
 
 type optsConfig struct {
-	idOpts             []func(*flow.Identity)
-	dhtPrefix          string
-	dhtOpts            []dht.Option
-	peerUpdateInterval time.Duration
-	connectionGating   bool
+	idOpts                       []func(*flow.Identity)
+	dhtPrefix                    string
+	dhtOpts                      []dht.Option
+	connectionGating             bool
+	streamsRateLimiter           unicast.RateLimiter
+	bandwidthRateLimiter         unicast.RateLimiter
+	onUnicastRateLimitedPeerFunc func(peerID peer.ID)
```

Comment:
it looks like this trio appears quite often together. Please consider wrapping this with a structure.

Reply:
https://github.com/onflow/flow-go/pull/2840/commits/103bdb60e540464910dca5a5ae459883f2adedc4